e2e/e2e_test.go

```diff
@@ -0,0 +1,150 @@
+package e2e
+
+import (
+	"context"
+	"github.com/celestiaorg/celestia-app/v4/app"
+	celestiadockertypes "github.com/celestiaorg/tastora/framework/docker"
+	"github.com/celestiaorg/tastora/framework/testutil/toml"
+	celestiatypes "github.com/celestiaorg/tastora/framework/types"
+	"github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/moby/moby/client"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+	"os"
+	"testing"
+)
+
+const (
+	celestiaAppImage   = "ghcr.io/celestiaorg/celestia-app"
+	defaultCelestiaTag = "v4.0.0-rc6"
+	nodeImage          = "ghcr.io/celestiaorg/celestia-node"
+	defaultNodeTag     = "v0.23.0-rc0"
+	testChainID        = "test"
+)
+
+func TestCelestiaTestSuite(t *testing.T) {
+	suite.Run(t, new(CelestiaTestSuite))
+}
+
+type CelestiaTestSuite struct {
+	suite.Suite
+	logger  *zap.Logger
+	client  *client.Client
+	network string
+}
+
+func (s *CelestiaTestSuite) SetupSuite() {
+	s.logger = zaptest.NewLogger(s.T())
+	s.logger.Info("Setting up Celestia test suite")
+	s.client, s.network = celestiadockertypes.DockerSetup(s.T())
+}
+
+// appOverrides modifies the "app.toml" configuration for the application, setting the transaction indexer to "kv".
+func appOverrides() toml.Toml {
+	// required to query tx by hash when broadcasting transactions.
+	appTomlOverride := make(toml.Toml)
+	txIndexConfig := make(toml.Toml)
+	txIndexConfig["indexer"] = "kv"
+	appTomlOverride["tx-index"] = txIndexConfig
+	return appTomlOverride
+}
+
+// configOverrides modifies the "config.toml" configuration, enabling KV indexing for transaction queries.
+func configOverrides() toml.Toml {
+	// required to query tx by hash when broadcasting transactions.
+	overrides := make(toml.Toml)
+	txIndexConfig := make(toml.Toml)
+	txIndexConfig["indexer"] = "kv"
+	overrides["tx_index"] = txIndexConfig
+	return overrides
+}
+
+func (s *CelestiaTestSuite) CreateDockerProvider() celestiatypes.Provider {
+	numValidators := 1
```

Comment:
since this is a correctness test, we can just run a single node, if there is a need/desire to increase this, it is always an option.